src/network_controller.py

```diff
@@ -51,12 +51,18 @@ def message_carry_request(self, message: Message):
         session = requests.Session()
         session.trust_env = False
         # send request to the network manager to see if in the same network
-        request = session.prepare_request(requests.Request('GET', "127.0.0.1:33000/message_in_range", params={
+        request = session.prepare_request(requests.Request('GET', f"{self._simulator_address}/{self._simulator_port}/message_in_range", params={
             "source_id": source_id,
             "destination_id": destination_id,
             "contents": contents
         }))
         res = requests.send(request)
+        if (res == 'Can send message'):
```

Comment:
Did you test this? I think it might be res.content, and you might consider using a json encoded boolean instead of a literal string, although if it works I dont have any problem with it.

Reply:
Sorry for bad quality coding ! Now I have modified and tested the code on local environment, and will upload later.